homework/tests/example.spec.js

```diff
@@ -27,5 +27,130 @@ data.forEach(version => {
+  
+      await expect(page.locator('#numberAnswerField')).toHaveValue('23');
+    });
+
+    test('Dividing 2 and 0 results in an error', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('2');
+      await page.locator('#number2Field').type('0');
+      await page.selectOption('#selectOperationDropdown', {label: 'Divide'});
+      await page.locator('#calculateButton').click();
+  
+      await expect(page.locator('#errorMsgField')).toHaveText('Divide by zero error!');
+    });
+
+    test('Clear button is enabled', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#clearButton').click();
+    });
+
+    test('Clear button clears the answer', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('1');
+      await page.locator('#number2Field').type('2');
+      await page.selectOption('#selectOperationDropdown', {label: 'Add'});
+      await page.locator('#calculateButton').click();
+      await page.locator('#clearButton').click();
+  
+      await expect(page.locator('#numberAnswerField')).toHaveValue('');
+    });
+
+    test('Integer only option is enabled', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#integerSelect').click();
+    });
+
+    test('Integer only option is changes answer to integer', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('1.1');
+      await page.locator('#number2Field').type('1.2');
+      await page.selectOption('#selectOperationDropdown', {label: 'Add'});
+      await page.locator('#calculateButton').click();
+      await expect(page.locator('#numberAnswerField')).toHaveValue('2.3');
+
+      await page.locator('#integerSelect').click();
+      await expect(page.locator('#numberAnswerField')).toHaveValue('2');
+
+    });
+
+    test('Adding [a] and 2 displays an error', async ({ page }) => {
+      await page.goto('https://testsheepnz.github.io/BasicCalculator');
+      await page.selectOption('#selectBuild', { label: version});
+      await page.locator('#number1Field').type('a');
+      await page.locator('#number2Field').type('2');
+      await page.selectOption('#selectOperationDropdown', {label: 'Add'});
+      await page.locator('#calculateButton').click();
+  
+      await expect(page.locator('#errorMsgField')).toHaveText('Number 1 is not a number');
```

Comment:
This function is used several times in tests. Please extract this to a separate function e.g. `AssertErrorMessage(message)`